walk/config.go

```diff
@@ -108,33 +107,42 @@ func (wc *Configurer) CheckFlags(_ *flag.FlagSet, c *config.Config) error {
 		}
 	}
 
-	c.Exts[walkName] = &walkConfig{
-		excludes: wc.cliExcludes,
-	}
 	return nil
 }
 
 func (*Configurer) KnownDirectives() []string {
 	return []string{"build_file_name", "generation_mode", "exclude", "follow", "ignore"}
 }
 
-func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
-	wc := getWalkConfig(c)
-	wcCopy := &walkConfig{}
-	*wcCopy = *wc
-	wcCopy.ignore = false
+func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {}
+
+func (wcp *walkConfig) newChild(rel string, f *rule.File) *walkConfig {
+	// MUST BE THREAD SAFE - will be invoked concurrently while fs-walking and BUILD-parsing.
+
+	wc := &walkConfig{
+		// Always false by default.
+		ignore: false,
+
+		// Ineritied, potentially overwritten with directives.
+		updateOnly:          wcp.updateOnly,
+		validBuildFileNames: wcp.validBuildFileNames,
+
+		// Will potentially be extended with directives, must fully clone.
```

Comment:
Optimization nit: you don't need to fully clone these. `wcp.excludes[:len(wcp.excludes):len(wcp.excludes)]` creates a slice that shares the same memory, but the capacity equals the length, so if anything's appended, it's guaranteed to copy the elements into a new backing array. Copy-on-append basically.

Reply:
Great 👍